src/binaryutils.cpp

```diff
@@ -265,6 +279,34 @@ InferenceEngine::Blob::Ptr convertMatsToBlob(std::vector<cv::Mat>& images, const
     }
 }
 
+Status convertBlobToMats(std::vector<cv::Mat>& images, const InferenceEngine::Blob::Ptr& blob, const std::shared_ptr<TensorInfo>& tensorInfo) {
+    int matPrecision = getMatTypeFromTensorPrecision(tensorInfo->getPrecision());
+    int rows, cols, channels;
+    if (tensorInfo->getLayout() == InferenceEngine::Layout::NCHW) {
+        channels = tensorInfo->getShape()[1];
+        rows = tensorInfo->getShape()[2];
+        cols = tensorInfo->getShape()[3];
+    } else if (tensorInfo->getLayout() == InferenceEngine::Layout::NHWC) {
```

Comment:
Do we want different behavior in binary inputs and binary outputs?
For binary inputs we enforce user to switch model to NHWC layout. We do not proceed further if model is in wrong layout. For binary outputs we allow NCHW?

NCHW will not even work here since you do not transpose data before converting to cv::Mat.